core/templates/components/button-directives/create-activity-button.component.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the Create Exploration/Collection button.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ExplorationCreationService } from 'components/entity-creation-services/exploration-creation.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { UserService } from 'services/user.service';
+import { CreateActivityModalComponent } from 'pages/creator-dashboard-page/modal-templates/create-activity-modal.component';
+import { AppConstants } from 'app.constants';
+
+ @Component({
+   selector: 'oppia-create-activity-button',
+   templateUrl: './create-activity-button.component.html',
+ })
+export class CreateActivityButtonComponent implements OnInit {
+   creationInProgress: boolean = false;
+   canCreateCollections: boolean = false;
+   allowYamlFileUpload: boolean;
```

Comment:
Should we provide a default?

Reply:
It depends on this constant `ALLOW_YAML_FILE_UPLOAD` basically,I'm making it false for now